src/lazy/expanded/mod.rs

```diff
@@ -215,12 +238,7 @@ pub struct ExpandingReader<Encoding: Decoder, Input: IonInput> {
     // Holds information found in symbol tables and encoding directives (TODO) that can be applied
     // to the encoding context the next time the reader is between top-level expressions.
     pending_lst: UnsafeCell<PendingLst>,
-    // A bump allocator that is cleared between top-level expressions.
-    allocator: UnsafeCell<BumpAllocator>,
-    // TODO: Make the symbol and macro tables traits on `Encoding` such that they can be configured
-    //       statically. Then 1.0 types can use `Never` for the macro table.
-    symbol_table: UnsafeCell<SymbolTable>,
-    macro_table: UnsafeCell<MacroTable>,
+    encoding_context: UnsafeCell<EncodingContext>,
```

Comment:
🗺️ Grouping the `allocator`, `symbol_table`, and `macro_table` into the `encoding_context` field meant that there were a handful of places I could simplify the `UnsafeCell` code in the expanding reader.